File: modules/_foundation/src/main/java/com/projecthub/base/workflow/api/WorkflowController.java
package com.projecthub.base.workflow.api;

import com.projecthub.base.workflow.api.dto.WorkflowDefinitionDTO;
import com.projecthub.base.workflow.api.dto.WorkflowTransitionRequestDTO;
import com.projecthub.base.workflow.api.mapper.WorkflowMapper;
import com.projecthub.base.workflow.application.service.WorkflowManagementService;
import com.projecthub.base.workflow.domain.model.WorkflowContext;
import com.projecthub.base.workflow.domain.model.WorkflowDefinition;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/api/v1/workflows")
@RequiredArgsConstructor
public class WorkflowController {
    private final WorkflowManagementService workflowService;
    private final WorkflowMapper workflowMapper;

    @GetMapping
    public ResponseEntity<List<WorkflowDefinitionDTO>> getAllWorkflows() {
        List<WorkflowDefinition> workflows = workflowService.getAllWorkflows();
        return ResponseEntity.ok(workflowMapper.toDtoList(workflows));
    }

    @GetMapping("/{id}")
    public ResponseEntity<WorkflowDefinitionDTO> getWorkflow(@PathVariable UUID id) {
        WorkflowDefinition workflow = workflowService.getWorkflowById(id);
        return ResponseEntity.ok(workflowMapper.toDto(workflow));
    }

    @PostMapping
    public ResponseEntity<WorkflowDefinitionDTO> createWorkflow(@RequestBody WorkflowDefinitionDTO workflowDto) {
        WorkflowDefinition workflow = workflowMapper.toEntity(workflowDto);
        WorkflowDefinition created = workflowService.createWorkflow(workflow);
        return ResponseEntity.ok(workflowMapper.toDto(created));
    }

    @PostMapping("/{workflowId}/transitions")
    public ResponseEntity<Void> executeTransition(
            @PathVariable UUID workflowId,
            @RequestBody WorkflowTransitionRequestDTO request) {
        WorkflowContext context = workflowMapper.toContext(request);
        workflowService.executeTransition(workflowId, context);
        return ResponseEntity.ok().build();
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteWorkflow(@PathVariable UUID id) {
        workflowService.deleteWorkflow(id);
        return ResponseEntity.ok().build();
    }
}

File: src/main/kotlin/com/projecthub/infrastructure/adapter/in/web/WorkflowController.kt
package com.projecthub.infrastructure.adapter.in.web

import com.projecthub.infrastructure.adapter.in.dto.WorkflowDefinitionDTO
import com.projecthub.infrastructure.adapter.in.dto.WorkflowTransitionRequestDTO
import com.projecthub.infrastructure.adapter.in.mapper.WorkflowMapper
import com.projecthub.core.application.service.WorkflowManagementService
import com.projecthub.core.domain.model.workflow.WorkflowContext
import com.projecthub.core.domain.model.workflow.WorkflowDefinition
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.util.UUID

@RestController
@RequestMapping("/api/v1/workflows")
class WorkflowController(
    private val workflowService: WorkflowManagementService,
    private val workflowMapper: WorkflowMapper
) {
    @GetMapping
    fun getAllWorkflows(): ResponseEntity<List<WorkflowDefinitionDTO>> {
        val workflows = workflowService.getAllWorkflows()
        return ResponseEntity.ok(workflowMapper.toDtoList(workflows))
    }

    @GetMapping("/{id}")
    fun getWorkflow(@PathVariable id: UUID): ResponseEntity<WorkflowDefinitionDTO> {
        val workflow = workflowService.getWorkflowById(id)
        return ResponseEntity.ok(workflowMapper.toDto(workflow))
    }

    @PostMapping
    fun createWorkflow(@RequestBody workflowDto: WorkflowDefinitionDTO): ResponseEntity<WorkflowDefinitionDTO> {
        val workflow = workflowMapper.toEntity(workflowDto)
        val created = workflowService.createWorkflow(workflow)
        return ResponseEntity.ok(workflowMapper.toDto(created))
    }

    @PostMapping("/{workflowId}/transitions")
    fun executeTransition(
        @PathVariable workflowId: UUID,
        @RequestBody request: WorkflowTransitionRequestDTO
    ): ResponseEntity<Void> {
        val context = workflowMapper.toContext(request)
        workflowService.executeTransition(workflowId, context)
        return ResponseEntity.ok().build()
    }

    @DeleteMapping("/{id}")
    fun deleteWorkflow(@PathVariable id: UUID): ResponseEntity<Void> {
        workflowService.deleteWorkflow(id)
        return ResponseEntity.ok().build()
    }
}
